CFC_DataCollector/modeinfer/experiment.py

```diff
@@ -0,0 +1,66 @@
+import scipy.io
+import random
+import numpy as np
+from sklearn.cross_validation import KFold
+
+mat = scipy.io.loadmat('original_data.mat')
+#modified_mat = scipy.io.loadmat('modified_data.mat')
+
+rows = mat['X'].shape[0]
+print rows
+mat['y'] = np.transpose(mat['y']).ravel()
+#test_size = 50
+test_size = 30
+
+orig_training_set = mat['X'][0:rows-test_size]
+print np.shape(orig_training_set)
+orig_training_labels = mat['y'][0:rows-test_size]
+
+orig_test_set = mat['X'][rows-test_size:]
+orig_test_labels = mat['y'][rows-test_size:]
+
+#mod_training_set = modified_mat['X'][0:rows-test_size]
+#mod_training_labels = modified_mat['y'][0:rows-test_size]
+
+#mod_test_set = modified_mat['X'][rows-test_size:]
+#mod_test_labels = modified_mat['y'][rows-test_size:]
+
+    
+def buildModelStep(selFeatureMatrix, cleanedResultVector):
+	from sklearn import ensemble
+	forestClf = ensemble.RandomForestClassifier()
+	model = forestClf.fit(selFeatureMatrix, cleanedResultVector)
+	return model
+	
+kf = KFold(rows, n_folds=5)
+scores = []
+for train, test in kf:
+    #model = buildModelStep(orig_training_set, orig_training_labels)
+    model = buildModelStep(mat['X'][train], mat['y'][train])
+    #new_model = buildModelStep(mat['X'], mat['y'].ravel())
+    
+    #predicted_labels = model.predict(orig_test_set)
+    predicted_labels = model.predict(mat['X'][test])
+    print(predicted_labels)
+    print(orig_test_labels)
+    
+    def score_classifier(predicted_labels, true_labels):
```

Comment:
This is not a great comparison because a high accuracy on the `walking` mode, for example, can pull up the overall accuracy and mask the really low accuracy of the motorized modes. That is why we used stratified k-fold validation for our estimation.

I have an ipython file that implements stratified k-fold validation and generates a confusion matrix. Let me see if I can find it...